src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -50,21 +50,28 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
     char out[UCT_IB_MLX5DV_ST_SZ_BYTES(create_qp_out)]         = {};
     char in_2init[UCT_IB_MLX5DV_ST_SZ_BYTES(rst2init_qp_in)]   = {};
     char out_2init[UCT_IB_MLX5DV_ST_SZ_BYTES(rst2init_qp_out)] = {};
+    uct_ib_mlx5_mmio_mode_t mmio_mode;
     uct_ib_mlx5_devx_uar_t *uar;
     int max_tx, max_rx, len_tx, len;
     int wqe_size;
     int dvflags;
     void *qpc;
     int ret;
 
-    uct_ib_iface_fill_attr(iface, attr);
+    uct_ib_iface_fill_attr(iface, &attr->super);
+
+    status = uct_ib_mlx5_get_mmio_mode(iface->super.worker, attr->mmio_mode,
```

Comment:
can we make setting mmio mode part of uct_rc_mlx5_iface_fill_attr()?

Reply:
not really
fill_attr goes before qp create
while in verbs mmio mode set after qp create